subsys/bluetooth/audio/cap_commander.c

```diff
@@ -77,17 +81,240 @@ int bt_cap_commander_discover(struct bt_conn *conn)
+		const union bt_cap_set_member *member = &param->param[i].member;
+		const struct bt_cap_common_client *client =
+			bt_cap_common_get_client(param->type, member);
+
+		if (client == NULL) {
+			LOG_DBG("Invalid param->param[%zu].member", i);
+			return false;
+		}
+
+		for (size_t j = 0U; j < i; j++) {
+			const union bt_cap_set_member *other = &param->param[j].member;
+
+			if (other == member) {
+				LOG_DBG("param->members[%zu] (%p) is duplicated by "
+					"param->members[%zu] (%p)",
+					j, other, i, member);
+				return false;
+			}
+		}
+
+		CHECKIF(start_param->num_subgroups == 0) {
+			LOG_DBG("param->param[%d]->num_subgroups is 0", i);
+			return false;
+		}
+
+		CHECKIF(start_param->subgroups == NULL) {
+			LOG_DBG("param->param[%d]->subgroup is NULL", i);
+			return false;
+		}
+	}
+
+	return true;
+}
+
 int bt_cap_commander_broadcast_reception_start(
 	const struct bt_cap_commander_broadcast_reception_start_param *param)
 {
-	return -ENOSYS;
+	const struct bt_cap_commander_proc_param *proc_param;
+	struct bt_cap_common_proc *active_proc;
+	struct bt_conn *conn;
+	struct bt_bap_broadcast_assistant_add_src_param add_src_param = {0};
+	int err;
+
+	if (bt_cap_common_proc_is_active()) {
+		LOG_DBG("A CAP procedure is already in progress");
+
+		return -EBUSY;
+	}
+
+	if (!valid_broadcast_reception_start_param(param)) {
+		return -EINVAL;
+	}
+
+	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_BROADCAST_RECEPTION_START, param->count);
+
+	broadcast_assistant_cb.add_src = cap_commander_ba_add_src_cb;
+	if (!ba_cb_registered && cap_commander_register_ba_cb() != 0) {
+		LOG_DBG("Failed to register VCP callbacks");
+
+		return -ENOEXEC;
+	}
+
+	active_proc = bt_cap_common_get_active_proc();
+
+	for (size_t i = 0U; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_start_member_param
+			*member_param = &param->param[i];
+		struct bt_conn *member_conn =
+			bt_cap_common_get_member_conn(param->type, &member_param->member);
+
+		if (member_conn == NULL) {
+			LOG_DBG("Invalid param->members[%zu]", i);
+
+			return -EINVAL;
+		}
+
+		/* Store the necessary parameters as we cannot assume that the supplied parameters
+		 * are kept valid
+		 * TODO: consider putting this into a function
+		 */
+		active_proc->proc_param.commander[i].conn = member_conn;
+		bt_addr_le_copy(
+			&active_proc->proc_param.commander[i].broadcast_reception_start.addr,
+			&member_param->addr);
+		active_proc->proc_param.commander[i].broadcast_reception_start.adv_sid =
+			member_param->adv_sid;
+		active_proc->proc_param.commander[i].broadcast_reception_start.broadcast_id =
+			member_param->broadcast_id;
+		active_proc->proc_param.commander[i].broadcast_reception_start.pa_interval =
+			member_param->pa_interval;
+		active_proc->proc_param.commander[i].broadcast_reception_start.num_subgroups =
+			member_param->num_subgroups;
+		active_proc->proc_param.commander[i].broadcast_reception_start.subgroups =
+			member_param->subgroups;
```

Comment:
Might be worth to store `active_proc->proc_param.commander[i].broadcast_reception_start` in a local variable to make these lines shorter and more readable